src/main/java/Utils/Tables.java

```diff
@@ -1,24 +1,105 @@
 package Utils;
 
+import Models.Event.Log;
+import Models.User.DuplicateEmailPresentException;
+import Models.User.User;
+import Models.User.UserDao;
+
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
 import java.sql.Connection;
 import java.sql.PreparedStatement;
 import java.sql.SQLException;
 
+import static Utils.Passwords.createSaltAndHash;
+
 public class Tables {
     public static void createUserTable() {
         Connection con = Database.getInstance().getConnection();
         try {
-            PreparedStatement ps = con.prepareStatement("create table users(ID int generated always as identity, " +
+            PreparedStatement ps = con.prepareStatement("create table users(" +
+                    "ID int generated always as identity unique, " +
                     "firstName varchar(30) not null, " +
```

Comment:
Are we using `firstName` as the username people log in with? Perhaps we should make that clearer in a future PR, as I wouldn't normally say those terms are interchangeable?

Reply:
The user logs in with a combination of their email and password (though the PK for a user is still an auto-incremented integer). Emails are required to be unique and an error will be given if somebody tries to register with a duplicate email.

Reply:
I see! I got confused because the admin user that you created and mention at the top of the conversation has a username: admin which isn't an email.

Reply:
Yeah in the future the initial admin account will be configurable in the config file